src/query/service/src/interpreters/interpreter_copy.rs

```diff
@@ -270,21 +272,43 @@ impl CopyInterpreter {
             (build_res, plan.required_source_schema.clone(), files)
         };
 
-        build_local_append_data_pipeline(
-            ctx,
+        let insert_overwrite_option = build_append_data_pipeline(
+            ctx.clone(),
             &mut build_res.main_pipeline,
-            plan.clone(),
+            CopyPlanType::CopyIntoTablePlanOption(plan.clone()),
             source_schema,
-            to_table,
-            files,
+            to_table.clone(),
         )?;
+        // if it's replace mode, don't commit
```

Comment:
why `replace` mode is special here?

Reply:
replace into statement can use copy statement as its source, for this case, copy is just a data source. no need to commit